### PR TITLE
MGMT-5382 Include operators' installation progress as part of the total installation progress

### PR DESCRIPTION
**Pull request by jkilzi**
This PR tweaks the progress bar calculation in a way that the hosts progress account for 75% of the overall progress and the operators for the remaining 25%

### CODE REVIEW

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 1]
```diff
@@ -68,6 +69,19 @@ const getProgressPercent = (hosts: Host[] = []) => {
...
+    if (
+      (operator.operatorType === 'builtin' && operator.status === 'available') ||
+      (operator.operatorType === 'olm' && ['available', 'failed'].includes(operator.status || ''))
+    ) {
+      nextTotal += 1 / monitoredOperators.length;
+    }
+
+    return nextTotal;
+  }, 0);
+
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx:84]: This seems easier to understand:
```suggestion
export const getOperatorsProgressPercent = (monitoredOperators: MonitoredOperatorsList) => {
  const completeOperators = monitoredOperators.filter(
    (operator) =>
      (operator.operatorType === 'builtin' && operator.status === 'available') ||
      (operator.operatorType === 'olm' && ['available', 'failed'].includes(operator.status || '')),
  );
  return (completeOperators.length / monitoredOperators.length) * 100;
};

```
Also note `* 100` at the end to actually get a percentage.

**Review by jtomasek — CHANGES_REQUESTED** [review]

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 2]
```diff
@@ -220,36 +254,39 @@ const ClusterProgress: React.FC<ClusterProgressProps> = ({ cluster, minimizedVie
...
-                <HostProgress hosts={hosts} hostRole="worker" />
-              </FlexItem>
-            )}
+          </RenderIf>
+
+          <FlexItem>
+            <FinalizingProgress cluster={cluster} />
+          </FlexItem>
+
+          <RenderIf condition={monitoredOperators.length > 0}>
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx:281]: This progress item should render only if there are any olm operators being installed.
```suggestion
          <RenderIf condition={getOlmOperators(monitoredOperators).length > 0}>
```

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 3]
```diff
@@ -220,36 +254,39 @@ const ClusterProgress: React.FC<ClusterProgressProps> = ({ cluster, minimizedVie
...
             <FlexItem>
-              <FinalizingProgress cluster={cluster} />
+              <OperatorsProgressItem
+                operators={monitoredOperators.filter(({ operatorType }) => operatorType === 'olm')}
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx:284]:
```suggestion
                operators={getOlmOperators(monitoredOperators)}
```

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 4]
```diff
@@ -195,10 +220,19 @@ type ClusterProgressProps = {
...
+  const hostsProgressPercent = React.useMemo(() => Math.round(getProgressPercent(cluster.hosts)), [
+    cluster.hosts,
+  ]);
+  const operatorsProgressPercent = React.useMemo(
+    () => Math.round(getOperatorsProgressPercent(monitoredOperators)),
+    [monitoredOperators],
+  );
+  const label = getProgressLabel(
+    cluster,
+    hostsProgressPercent * 0.75 + operatorsProgressPercent * 0.25,
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx:232]: I think it would be better to do `Math.round` only here for better results

Comment (jtomasek) [reply src/components/clusterDetail/ClusterProgress.tsx:232]: With current calculation we're getting percentages such as this:
Screenshot 2021-05-13 at 12 47 04

**Comment by nirfarkas** [comment]
It's not awful, but a bit ugly though (: can we round it up somehow to show full numbers?

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 5]
```diff
@@ -195,10 +218,21 @@ type ClusterProgressProps = {
 
 const ClusterProgress: React.FC<ClusterProgressProps> = ({ cluster, minimizedView = false }) => {
   const { status, monitoredOperators = [] } = cluster;
-  const hosts = getEnabledHosts(cluster.hosts);
-  const progressPercent = React.useMemo(() => Math.round(getProgressPercent(hosts)), [hosts]);
-  const label = getProgressLabel(cluster, progressPercent);
-  const isWorkersPresent = hosts && hosts.some((host) => host.role === 'worker');
+  const hostsProgressPercent = React.useMemo(
+    () => Math.round(getHostsProgressPercent(cluster.hosts)),
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx:222]: Is the rounding still necessary here when we're doing it in the label?

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 6]
```diff
@@ -195,10 +218,21 @@ type ClusterProgressProps = {
...
+    [cluster.hosts],
+  );
+  const operatorsProgressPercent = React.useMemo(
+    () => Math.round(getOperatorsProgressPercent(monitoredOperators)),
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx:226]: And here

**Review by jtomasek — APPROVED** [review]
Great UX Improvement!

**Pull request merged**